inc/html.class.php

```diff
@@ -5322,7 +5322,10 @@ static function redefineAlert() {
       echo self::scriptBlock("
       window.old_alert = window.alert;
       window.alert = function(message, caption) {
-         message = message.replace('\\n', '<br>');
+         // Don't apply methods on undefined objects... ;-) #3866
+         if(message !== undefined) {
```

Comment:
i think this test could be better with: `message.length > 0` or `typeof message == "String"`
Test it !

Reply:
@orthagh undefined.anything is never a great idea ;-)
I update for the typeof comparision !